utils.cpp

```diff
@@ -693,28 +693,42 @@ char *get_file_contents (const char *filename) {
 	// find file size (Spencer says it's not a hack)
 	fseek (file, 0, SEEK_END);
 	size = ftell (file);
+
+	// don't panic if the file is empty
+	if (size == 0) {
+		read_size = 1;
+		size = 1;
+		buffer = (char*)  malloc (2);
+		buffer[0] = '\n';
+		goto finished_reading;
+	}
+
 	rewind (file);
 
 
 	// If there is a UTF-8 endian marker at the beginning of the file, skip it.

```

Comment:
Since the implementation of this is the issue, I think the problem can be fixed more easily by changing this only.

Something like..

```c
unsigned char file_head[3];
read_size = fread(file_head, 1, 3, file);
if (read_size == 3 && memcmp(file_head, utf8_endian_mark, sizeof(file_head)) == 0) {
    // Skip UTF-8 BOM
    size -= sizeof(file_head);
} else {
    // Read data needs to be retained
    rewind(file);
}
```

Reply:
Thanks, I will incorporate that into a new commit.